xarray/core/combine.py

```diff
@@ -323,16 +324,17 @@ def _dataarray_concat(arrays, dim, data_vars, coords, compat,
         raise ValueError('data_vars is not a valid argument when '
                          'concatenating DataArray objects')
 
-    datasets = []
-    for n, arr in enumerate(arrays):
-        if n == 0:
-            name = arr.name
-        elif name != arr.name:
-            if compat == 'identical':
-                raise ValueError('array names not identical')
-            else:
-                arr = arr.rename(name)
-        datasets.append(arr._to_temp_dataset())
+    name = result_name(arrays)
+    names = [arr.name for arr in arrays]
+    if compat == 'identical' and len(set(names)) != 1:
+        raise ValueError('array names not identical')
```

Comment:
Can you add a test for this error message? Also, please include the conflicting names in the error message -- that makes the error messages easier to debug.

Reply:
Done - the new message includes all names if ten or less, but just the unique names over ten to avoid incredibly long messages with large data.  Plus a test of course!